code/modules/projectiles/gun.dm

```diff
@@ -1087,14 +1129,14 @@ and you're good to go.
 			else
 				active_attachable.fire_attachment(target, src, user) //Fire it.
 				active_attachable.last_fired = world.time
-			return
+			return NONE
 			//If there's more to the attachment, it will be processed farther down, through in_chamber and regular bullet act.
 	/*
 	This is where burst is established for the proceeding section. Which just means the proc loops around that many times.
 	If burst = 1, you must null it if you ever RETURN during the for() cycle. If for whatever reason burst is left on while
 	the gun is not firing, it will break a lot of stuff. BREAK is fine, as it will null it.
 	*/
-	else if((flags_gun_features & GUN_BURST_ON) && burst_amount > 1)
+	else if((gun_firemode == GUN_FIREMODE_BURSTFIRE) && burst_amount > 1) //Zonenote: move to component?
```

Comment:
I think the fire mode could be moved into a component applied to the gun

Reply:
What's gained from that?